modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java

```diff
@@ -249,18 +273,40 @@ private static int getLogFileSequenceNumber() {
                         + "found " + attributeExpressionExecutors[1].getReturnType());
             }
 
-            if (attributeExpressionLength == 3) {
-                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+
+            if (attributeExpressionLength >= 3) {
+
+                if (!(attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor)) {
+                    throw new SiddhiAppValidationException("windowSize has to be constant but " +
+                            "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
+                }
+
+                if (!(attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT)) {
+                    throw new SiddhiAppValidationException("windowSize expected to be long but "
+                            + "found " + attributeExpressionExecutors[2].getReturnType());
+                } else {
                     recordWindow = (int) ((ConstantExpressionExecutor) attributeExpressionExecutors[2]).getValue();
+                }
+            }
+
+            if (attributeExpressionLength == 4) {
+
+                if (!(attributeExpressionExecutors[3] instanceof ConstantExpressionExecutor)) {
+                    throw new SiddhiAppValidationException("ID has to be constant but " +
+                            "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
```

Comment:
here this.attributeExpressionExecutors[1].getClass().getCanonicalName()); should be changed to

this.attributeExpressionExecutors[3].getClass().getCanonicalName());

Reply:
Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df